### ISSUE TITLE
lxc: breaking changes between version 5.21.0 and 6.0.1

### ISSUE THREAD

**Issue by voidzero** [NixOS/nixpkgs#338793]
### Describe the bug
After LXC was updated from 5.21.0 in channel 24.05, there are a number of changes that break how containers are started and this is a breaking change I didn't expect to happen in the stable channel.

lxc start: when trying to start my container ("mx") I now get the following:

```
# lxc start mx
lxc-start: mx: ../src/lxc/tools/lxc_start.c: lxc_start_main: 267 No container config specified
```

I don't know where the config file lives (or used to live) so I don't know how to fix this. Apparently they took out `lxc config edit` so that no longer works either:

```
# lxc config edit mx
Unknown configuration item: edit
```

### Notify maintainers
@aanderse @adamcstephens @jnsgruk @megheaiulian @mkg20001

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

Note: this is where everything still works - I can't have the system running on the newer version because I need this mx container up and running for email.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
# nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.8.11-hardened1, NixOS, 24.05 (Uakari), 24.05.20240701.d032c1a`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.4`
 - channels(markvd): `""`
 - nixpkgs: `/etc/nix/path/nixpkgs`
```

The version I tried where lxc fails is `24.05.20240830.6e99f2a`

Thank you.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by adamcstephens** [comment]
Can you provide relevant configuration?

**Comment by voidzero** [comment]
Hi @adamcstephens sure, but I'm not certain which config?

Here is how I install lxc and lxd from my flake:

```
  virtualisation = {
    docker = {
      enable = true;
      logDriver = "syslog";
    };

    lxd = {
      enable = true;
      recommendedSysctlSettings = true;
      package = pkgs.unstable.lxd-lts;
      lxcPackage = pkgs.unstable.lxc;
    };

    lxc.lxcfs.enable = true;

    libvirtd = {
      enable = true;
      qemu = {
        package = pkgs.qemu_kvm;
        swtpm.enable = true;
        verbatimConfig = ''
          user = "markvd"
          group = "kvm"
          namespaces = []
        '';
      };
      allowedBridges = [ "vbr0" "vbr1" ];
    };
  };

```

I don't have /etc/lxc nor /etc/lxd. Maybe you can point me into the right direction?

**Comment by adamcstephens** [comment]
That's the important config, thanks. So you're using `lxd`, which confusingly uses the `lxc` command as a client. You appear to be running the `lxc` command from the `lxc` package, and *not* the one from the `lxd` package which you probably want.

Do you include `lxc` in your `environment.systemPackages` somewhere else, or perhaps mistakenly enable `virtualisation.lxc` (not needed for lxd) ? [That's not done](https://github.com/NixOS/nixpkgs/blob/fd9bfea989098bb9f05670e189f452e17f254007/nixos/modules/virtualisation/lxd.nix#L146) by `virtualisation.lxd` so it's ending up in your path somehow.

**Comment by voidzero** [comment]
Sorry for being slow on this - I'm dealing with some health issues.
You were right - I did have lxc in my environment.systemPackages. Removing that fixed the issue and we're running on 24.05 again smoothly.
